src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/ActivitiesResource.kt

```diff
@@ -200,20 +200,20 @@ class ActivitiesResource(
   fun createOffenderProgramProfile(
     @Schema(description = "Course activity id", required = true) @PathVariable courseActivityId: Long,
     @RequestBody @Valid
-    createRequest: CreateOffenderProgramProfileRequest,
-  ): OffenderProgramProfileResponse =
-    allocationService.createOffenderProgramProfile(courseActivityId, createRequest)
+    createRequest: CreateAllocationRequest,
+  ): CreateAllocationResponse =
+    allocationService.createAllocation(courseActivityId, createRequest)
 
   @PreAuthorize("hasRole('ROLE_NOMIS_ACTIVITIES')")
-  @PutMapping("/activities/{courseActivityId}/booking-id/{bookingId}/end")
+  @PutMapping("/activities/{courseActivityId}/allocations")
```

Comment:
Turn this into a generic update allocation endpoint to be consistent with the create endpoint